app/Jobs/RefreshNftMetadata.php

```diff
@@ -40,40 +43,48 @@ public function __construct(
      */
     public function handle(AlchemyWeb3DataProvider $provider): void
     {
-
         if (SpamContract::isSpam($this->collection->address, $this->collection->network)) {
```

Comment:
This call should be moved to the controller to prevent queuing metadata requests for unrelated collections.